MoreLinq.Test/DuplicatesTest.cs

```diff
@@ -0,0 +1,140 @@
+
+            using var source = TestingSequence.Of(Source());
+
+            void Act() => source.Duplicates().Take(1);
+
+            Assert.That(Act, Throws.Nothing);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_Without_Duplicates_Then_Empty_Sequence_Is_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates();
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of(
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement"
+                );
+
+            var duplicates = testingSequence.Duplicates().ToArray();
+
+            Assert.That(duplicates, Contains.Item("DUPLICATED_STRING"));
+            Assert.That(duplicates.AtMost(1), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Multiple_Duplicates_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of(
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement",
+                "SECOND_DUPLICATED_STRING",
+                "SECOND_DUPLICATED_STRING"
+            );
+
+            var duplicates = testingSequence.Duplicates().ToArray();
+
+            Assert.That(duplicates, Contains.Item("DUPLICATED_STRING"));
+            Assert.That(duplicates, Contains.Item("SECOND_DUPLICATED_STRING"));
+            Assert.That(duplicates.AtMost(2), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_True_Comparer_Then_Duplicates_Are_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates(new DummyStringAlwaysTrueComparer()).ToArray();
+
+            Assert.That(duplicates.AtMost(1), Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_False_Comparer_Then_Empty_Sequence_Is_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates(new DummyStringAlwaysFalseComparer());
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Multiple_Duplicates_Sequence_With_Custom_Always_False_Comparer_Then_Empty_Sequence_Is_Returned()
```

Comment:
To me, this test is redundant with the other two:

- `When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_False_Comparer_Then_Empty_Sequence_Is_Returned`
- `When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_True_Comparer_Then_Duplicates_Are_Returned`

What you really want to test here is (and the name should be updated to reflect that) is that `Duplicates` uses the custom comparer and the body of the tests sufficiently covers that. The other two tests don't go further to prove the point by changing the input or reverting the result of the comparer. Unless you see a flaw in my argument, let's just keep this one.

Reply:
i'll keep the false one as it find it more explicit than the always truth one